test/integration/modules/recipe_preferences/RecipePreferencesResolver.spec.ts

```diff
@@ -0,0 +1,135 @@
+import testServer, { loadConfig } from "../../../__utils/testServer";
+import { Container } from "typedi";
+import { DATABASE, IDatabase } from "../../../../src/Database";
+import { v4 as uuidv4 } from "uuid";
+import { User } from "../../../../src/model/User";
+import {
+  RECIPE_PREFERENCES_PROVIDER,
+  RecipePreferencesProvider,
+} from "../../../../src/modules/recipe_preferences/RecipePreferencesProvider";
+import {
+  USER_PROVIDER,
+  UserProvider,
+} from "../../../../src/modules/user/UserProvider";
+import { RecipePreferences } from "../../../../src/model/RecipePreferences";
+
+let query;
+let mutation;
+let mock_user1: User;
+
+const recipePreferencesForUserQuery = `query {
+  recipePreferencesForUser {
+    vegan
+    vegetarian
+    gluten
+    dairy
+  }
+}`;
+
+const setRecipePreferencesMutation = `mutation($prefs: RecipePreferencesInput!) {
+  setRecipePreferencesForUser(preferences: $prefs) {
+    vegan
+    vegetarian
+    gluten
+    dairy
+  }
+}`;
+
+beforeAll(async () => {
+  loadConfig();
+
+  const userProvider = Container.get(USER_PROVIDER) as UserProvider;
+  mock_user1 = await userProvider.createUserWithUUID(uuidv4());
+
+  const server = await testServer({
+    context() {
+      return {
+        user: mock_user1,
+      };
+    },
+  });
+  query = server.query;
+  mutation = server.mutate;
+});
+
+afterAll(async () => {
+  await (Container.get(DATABASE) as IDatabase).close();
+});
+
+describe("QUERY get RecipePreferences", () => {
+  let mock_preferences: RecipePreferences;
+
+  beforeAll(async () => {
```

Comment:
Ich glaube, das müsste `beforeEach` sein, oder? Sonst hast Du zwei Mal `beforeAll` definiert 😅

Reply:
Das ist schon so richtig. Dieses `beforeAll` ist innerhalb des `describe` und gilt dadurch nur für die `it` die innerhalb vom `describe` ausgeführt werden.